src/dataEntities/startStop.ts

```diff
@@ -38,10 +38,10 @@ export abstract class StartStopService extends EventEmitter {
     /**
```

Comment:
I agree start and stop sthould be async, but one change is needed:
- As it is async, nothing prevents different objects to call `start()` again before `startInternal()` completes, so that `mStarted` is still false. 

While this is unrelated to this commit, I am thinking of this other change:
- As it could make sense to share the same service in two other objects (e.g. two different objects might want to use the same ReorgDetector), it could make sense to change the semantic from start/stop to something like attach/detach. That is, the service actually stops when everyone detaches.

Reply:
Added an mStarting var, and a test to ensure it works.